src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs

```diff
@@ -106,18 +103,18 @@ private static MethodInfo FindMethod(Type startupType, string methodName, string
             var methodNameWithEnv = string.Format(CultureInfo.InvariantCulture, methodName, environmentName);
             var methodNameWithNoEnv = string.Format(CultureInfo.InvariantCulture, methodName, "");
 
-            var methods = startupType.GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
+            var methods = startupType.GetMethods(BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
```

Comment:
This will cause new conflicts on line 108 if you have a public and a private overload of the method. How about just changing the original exceptions to "A public method named..."